feature added #12, clickable links on output

sudipt1999 opened the pull request:
What does this PR fixes/adds
1. Added the feature clickable links on output
2. Added styles to them
3. Createad a generic function for output purpose that can be used in all types of output but currently is added to GET request only (Kindly let me know this one! I will be more than happy to add them).

Working proof is added here 
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/38929617/77018257-e4a72880-69a2-11ea-8971-32922d00b423.gif)

xadahiya commented on console-frontend/src/components/hydra-console/HydraConsole.jsx:
```diff
@@ -305,8 +367,13 @@ class HydraConsole extends React.Component {
               .then( (response) =>  {
+                let outputText = ""
+                if(response.data.length === 0)
+                 outputText = String(JSON.stringify(response.data, this.jsonStringifyReplacer, 8))
```
If response.data.length is 0, why would you stringify an empty string?

sudipt1999 replied:
Sure @xadahiya , actually it was because I was rendering the jsx incase its length was >0 so I stringify in case of empty response but thanks for correcting I will make changes accordingly

xadahiya commented on console-frontend/src/components/hydra-console/HydraConsole.jsx:
```diff
@@ -276,7 +291,54 @@ class HydraConsole extends React.Component {
+            resourcesIDs: resourcesIDs
+        })
+    }
+
```
Please add proper comments/docstrings to explain what each function does

xadahiya replied:
It would be great if you can do this for all the functions not just for the functions you wrote. :smile:

sudipt1999 replied:
Sure thing to do, my plan was to add them to every function once it was reviewed! Thank you

xadahiya commented on console-frontend/src/components/hydra-console/HydraConsole.jsx:
```diff
@@ -276,7 +291,54 @@ class HydraConsole extends React.Component {
+                arr.push(<div className={classes.outputEntryDiv}></div>)
+                    arr.push(<span className={classes.outputConsoleKey}>{key} :</span>)
+                    if(key == "@id")
+                    arr.push(<span className={classes.outputConsoleLink}
```
You should consider adding {} to improve readability and futureproofing (in case we want to add something else)

sudipt1999 commented:
@xadahiya thanks for reviewing my work ! Will update with changes soon

sudipt1999 commented:
@xadahiya so I have added generic methos to support any kind of json response, and added the method to all the request response as asked, kindly review it, a proof of attachment from my side has been added, all the elements are jsx instead of string like earlier  :
![Screenshot from 2020-03-22 01-13-17](https://user-images.githubusercontent.com/38929617/77235407-5ff31f00-6bdb-11ea-9086-19d54adc5517.png)
![Screenshot from 2020-03-22 01-12-29](https://user-images.githubusercontent.com/38929617/77235408-62ee0f80-6bdb-11ea-87c4-1de9aec73c0b.png)
![Screenshot from 2020-03-22 01-11-43](https://user-images.githubusercontent.com/38929617/77235409-66819680-6bdb-11ea-8140-170724fc8f9b.png)
![Screenshot from 2020-03-22 01-11-25](https://user-images.githubusercontent.com/38929617/77235411-697c8700-6bdb-11ea-8548-4238a97e3480.png)
